core/src/main/scala/kafka/log/LogSegment.scala

```diff
@@ -601,8 +601,8 @@ class LogSegment private[log] (val log: FileRecords,
     if (_maxTimestampSoFar.nonEmpty || _offsetOfMaxTimestampSoFar.nonEmpty) {
       CoreUtils.swallow(timeIndex.maybeAppend(maxTimestampSoFar, offsetOfMaxTimestampSoFar, skipFullCheck = true), this)
     }
-    CoreUtils.swallow(offsetIndex.close(), this)
-    CoreUtils.swallow(timeIndex.close(), this)
+    CoreUtils.swallow(lazyOffsetIndex.getLazy.foreach(_.close()), this)
```

Comment:
How about implement a close function in the LazyOffsetIndex and LazyTimeIndex class?